[Docs] Restructure doc page and add AI gallery (#3331)
skypilot-org/skypilot: docs/source/_gallery_original/index.rst

AI Gallery
====================

> AI Gallery is a collection of popular AI frameworks, LLM models, and Tutorials.
> It aims to speed up the evolution of AI by providing a simple way to **package**, **share**, and **distribute** AI using the simple interface of SkyPilot.
> Readers can directly take the YAML files and use them to run AI on their own infrastructure, such as clouds or Kubernetes.

Reviewer: ```suggestion
AI Gallery is a collection of ready-to-run recipes for popular AI frameworks and AI models.
It provides a simple way to **package**, **share**, and **distribute** AI projects using the simple interface of SkyPilot.

Readers can directly execute the SkyPilot YAMLs on their own infrastructure, such as cloud VMs or Kubernetes.
```

.. image:: https://imgur.com/dA0Llxh.png
   :alt: AI Gallery
   :align: center

Contents
--------

.. toctree::
   :maxdepth: 1
   :caption: Inference Engine

   vLLM <frameworks/vllm>
   SGLang <frameworks/sglang>
   Hugging Face TGI <frameworks/tgi>
   LoRAX <frameworks/lorax>

.. toctree::
   :maxdepth: 1
   :caption: LLM Models
   
   Mixtral (Mistral.ai) <llms/mixtral>

>    Mistral 7B (Mistral.ai) <https://docs.mistral.ai/self-deployment/skypilot/>

Author: Good point. Updated. Thanks!

   Llama-2 (Meta) <llms/llama-2>
   CodeLlama (Meta) <llms/codellama>
   Gemma (Google) <llms/gemma>

.. toctree::
   :maxdepth: 1
   :caption: Applications

>    Tabby: Code Assistant <applications/tabby>

Author: Updated. Thanks!

   LocalGPT: Chat with PDF <applications/localgpt>

.. toctree::
   :maxdepth: 1
   :caption: Tutorials

   tutorials/finetuning.md

> Contributing

Reviewer: ```suggestion
Adding an Example to the Gallery
```

------------
We welcome contributions from the community. If you would like to contribute, please follow the guidelines below.

1. Fork the `SkyPilot repository <https://github.com/skypilot-org/skypilot>`__ on GitHub.

> 2. Create a new folder for your own framework, LLM model, or Tutorial under `llm/ <https://github.com/skypilot-org/skypilot/tree/master/llm>`__.

Reviewer: ```suggestion
2. Create a new folder for your own framework, LLM model, or tutorial under `llm/ <https://github.com/skypilot-org/skypilot/tree/master/llm>`__.
```

> 3. Add your own README, SkyPilot YAML file and the necessary files to run your AI.

Reviewer: ```suggestion
3. Add a README.md, a SkyPilot YAML, and other necessary files to run the AI project.
```

4. Create a soft link in `docs/source/_gallery_original <https://github.com/skypilot-org/skypilot/blob/master/docs/source/_gallery_original>`__ to the README file in one of the subfolders (frameworks, llms, tutorials), e.g., :code:`cd docs/source/_gallery_original/llms; ln -s ../../llm/mixtral/README.md mixtral.md`.

> 5. Add the file path to the `toctree` above.

Reviewer: ```suggestion
5. Add the file path to the ``toctree`` above on this page.
```

AI Gallery
====================

AI Gallery is a collection of ready-to-run recipes for popular AI frameworks and AI models.

> It provides a simple way to **package**, **share**, and **distribute** AI projects using the simple interface of SkyPilot.
> 
> Readers can directly execute the SkyPilot YAMLs on their own infrastructure, such as cloud VMs or Kubernetes.

Reviewer: ```suggestion
It provides a simple way to **package**, **share**, and **run** AI projects using the simple interface of SkyPilot.

Readers can directly execute the recipes on their own infrastructure, such as cloud VMs (link to install) or Kubernetes (link to k8s page).
```
Does the distribute -> run change look good? "distribute" is covered by "share".

Author: Sounds good! Updated.

AI Gallery
====================

AI Gallery is a collection of ready-to-run recipes for popular AI frameworks and AI models.
It provides a simple way to **package**, **share**, and **run** AI projects using the simple interface of SkyPilot.

> Readers can directly execute the SkyPilot YAMLs on their own infrastructure, such as cloud VMs or Kubernetes.

Reviewer: ```suggestion
You can directly run these recipes on their own infrastructure, such as cloud VMs or Kubernetes.
```
This is to fit the sentence on 1 line (on laptop). Also, "run recipes" more accurate than running YAMLs.

Also: add the 2 links (install / k8s docs)?

Author: Good points! Fixed both.

AI Gallery is a collection of ready-to-run recipes for popular AI frameworks and AI models.
It provides a simple way to **package**, **share**, and **run** AI projects using the simple interface of SkyPilot.

Readers can directly execute the SkyPilot YAMLs on their own infrastructure, such as cloud VMs or Kubernetes.

> .. image:: https://imgur.com/POFdmiEx.png

Reviewer: N/A now? https://skypilot.readthedocs.io/en/docs-with-gallery-2/gallery/index.html Shall we use our own?

1. Fork the `SkyPilot repository <https://github.com/skypilot-org/skypilot>`__ on GitHub.
2. Create a new folder for your own framework, LLM model, or tutorial under `llm/ <https://github.com/skypilot-org/skypilot/tree/master/llm>`__.

> 3. Add a README.md, a SkyPilot YAML, and other necessary files to run the AI project. (Relative paths are not supported. Please use GitHub URLs for the references to the files.)

Reviewer: Was thinking this hint can exist  ~L16 as a comment for reviewers to pay attention to. To make this section shorter.
